emcc.py

```diff
@@ -511,7 +511,7 @@ def filter_emscripten_options(argv):
         src = open(target).read()
         full_node = ' '.join(shared.NODE_JS)
         if os.path.sep not in full_node:
-          full_node = '/usr/bin/' + full_node # TODO: use whereis etc. And how about non-*NIX?
+          full_node = '/usr/bin/env ' + full_node # TODO: use whereis etc. And how about non-*NIX?
```

Comment:
Perhaps this could just unconditionally call `shared.which(full_node)`? https://github.com/kripken/emscripten/blob/incoming/tools/shared.py#L1436 (in particular if `full_node` refers to an absolute path, that function already returns it unmodified)

Do you know what the extent of the `env` program is on different distros? Perhaps that could be avoided by reusing the above? Although if `NODE_JS` has cmdline params, I think those should not be passed to `shared.which()`

Reply:
>Perhaps this could just unconditionally call `shared.which(full_node)`?

nvm path depends on Node.js version. For example: `/home/saschanaz/.nvm/versions/node/v9.3.0/bin/node`. This would prevent users from using Node.js on their own paths if the build result is released publicly.

>Do you know what the extent of the `env` program is on different distros?

AFAIK `env` is in the UNIX specification. We already are using `#!/usr/bin/env python` on emcc so it shouldn't matter.